### PR TITLE
Sync Travis-CI configs

### PR DESCRIPTION
**Pull request by at-wat**
Common CI setting for the latest ~one~ two Go versions.

### CODE REVIEW

**Comment by Sean-Der** [comment]
We might be able to cover all repos now! https://stackoverflow.com/a/54919732 it looks like having to fetch Go just for WASM might not be true (just misconfigured path)

If we can setup a build matrix to do i386 + ARM + x64 for test (but x64 only for lint) I think that should cover everything.

File: travis-simple/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - name: test
+      env: CACHE_NAME=test
+      before_install:
+        - go mod download
+      install:
+        - go build ./...
+      script:
+        - coverpkgs=$(go list ./... | grep -v examples | paste -s -d ',')
+        - |
+          go test \
```
Comment (Sean-Der) [inline travis-simple/.travis.yml]: If tests deadlock this doesn't produce any `stdout` it is really frustrating. 

I haven't looked at in a while, but I need to look if we can do this a better way.

**Comment by Sean-Der** [comment]
We might not need to split for lint+test! If we can get everything standardized long term I think we will be in a much better spot.

I will see if I can `pion/webrtc` to not need its customization. It is the only repo I am aware of that does extra stuff.

**Comment by at-wat** [comment]
Sounds great!
A few repos are also excluded temporary.
- turn: golang-ci version is 1.19.1 which has some extra linter which fails on some other repos
- dtls: trying to test on the latest two Go version: https://github.com/pion/dtls/pull/159

Feel free to add commits to this branch.

File: travis-simple/.travis.yml (hunk 1 above)
Comment (at-wat) [reply travis-simple/.travis.yml]: `go test -timeout=${TIMEOUT_SECONDS}s` may help?

**Comment by at-wat** [comment]
> turn: golang-ci version is 1.19.1 which has some extra linter which fails on some other repos

It might be better to just move to 1.19 and disable new linters by `.golangci.yml` per repository until the repository addresses them.

**Comment by at-wat** [comment]
Fixing wasm test at https://github.com/pion/quic/pull/16

**Comment by at-wat** [comment]
`webrtc` pkg requires `wrtc` package.
Need to setup hook in `go_js_wasm_exec`.

**Comment by at-wat** [comment]
I have tried to use wasmbrowsertest and Chrome to remove test dependency to `wrtc`.
Test works fine, but browser can't output coverage report.
https://github.com/pion/webrtc/pull/973

File: travis-simple/.travis.yml (hunk 1 above)
Comment (at-wat) [reply travis-simple/.travis.yml]: `go test -timeout=` specifies total time of the tests. It's not suitable for our case.

**Comment by at-wat** [comment]
Running tests for WASM on all repos looks messy as WASM is still in Minimum Viable Product stage.
e.g. IP socket is not yet supported and `net_fake.go` is used in Go1.13. A lot of the tests using `net.Conn` fail to be built, deadlock, or cause runtime error due to a bug in `net_fake.go`.
```
panic: interface conversion: net.sockaddr is *net.UDPAddr, not *net.TCPAddr
```

File: ci/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      GOOS=js GOARCH=wasm go test \
+        -coverpkg=${coverpkgs} -coverprofile=cover.out -covermode=atomic \
+        -exec="${GO_JS_WASM_EXEC}" \
+        -v ./...
+  after_success:
+    - travis_retry bash <(curl -s https://codecov.io/bash) -c -F wasm
+
+jobs:
+  include:
+    - <<: *lint_job
```
Comment (daenney) [inline ci/.travis.yml]: I don't think we need this one, there doesn't seem to be much point to linting code with Go 1.12 and Go 1.13. I would suggest only having a single lint job, with the latest version of Go we support.

Comment (at-wat) [reply ci/.travis.yml]: sounds reasonable.

**Comment by at-wat** [comment]
It should be ready after merging WASM fix PRs.

**Comment by daenney** [comment]
I've gone ahead and approved them all, since they were just a build tag.

File: .github/workflows/assets-sync.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@v2
-      - name: sync
+      - name: sync lint scripts
```
Comment (at-wat) [inline .github/workflows/assets-sync.yml]:
```suggestion
      - name: sync CI configs
```

**Comment by at-wat** [comment]
commit from github web ui looks using CR+LF.

**Comment by at-wat** [comment]
Merging now. Will try syncing to the repos.

**Pull request merged**